tests/api_fastapi/execution_api/routes/test_task_instances.py

```diff
@@ -455,6 +455,36 @@ def test_ti_heartbeat_update(self, client, session, create_task_instance, time_m
         session.refresh(ti)
         assert ti.last_heartbeat_at == time_now.add(minutes=10)
 
+    def test_ti_update_state_to_failed_table_check(self, client, session, create_task_instance):
+        from math import ceil
+
+        ti = create_task_instance(
+            task_id="test_ti_update_state_to_terminal",
+            start_date=DEFAULT_START_DATE,
+            state=State.RUNNING,
+        )
+        ti.start_date = DEFAULT_START_DATE
+        session.commit()
+
+        response = client.patch(
+            f"/execution/task-instances/{ti.id}/state",
+            json={
+                "state": State.FAILED,
+                "end_date": DEFAULT_END_DATE.isoformat(),
+            },
+        )
+
+        assert response.status_code == 204
+        assert response.text == ""
+
+        session.expire_all()
+
+        ti = session.get(TaskInstance, ti.id)
+        assert ti.state == State.FAILED
+        assert ti.next_method is None
+        assert ti.next_kwargs is None
+        assert ceil(ti.duration) == 3600.00
```

Comment:
Why do we need `ceil`?

Reply:
`duration_expression_update` returns the value here as a decimal which cannot be compared for equality. See an example here: https://github.com/apache/airflow/actions/runs/12350551354/job/34464066950

It returns: `3599.999986588955`.

Reply:
@amoghrajesh Should be solved by https://github.com/apache/airflow/pull/45002

Reply:
Sqlite is only used for testing related activities right? Should we have to handle it?

Reply:
Got it, makes sense to handle it.

Reply:
Although, you don't have to wait for that PR to be merged, you can merge this and I can handle any chances in https://github.com/apache/airflow/pull/45002

Reply:
Alright, will leave a TODO